tests/test_unittest/test_gtscript_frontend.py

```diff
@@ -231,6 +232,129 @@ def definition_func(inout_field: gtscript.Field[float]):
             compile_definition(definition_func, "test_wrong_value", module, externals=externals)
 
 
+class TestIntervalSyntax:
+    def test_simple(self):
+        def definition_func(field: gtscript.Field[float]):
+            with computation(PARALLEL), interval(0, 1):
+                field = 0
+
+        module = f"TestIntervalSyntax_simple_{id_version}"
+        externals = {}
+        stencil_id, def_ir = compile_definition(
+            definition_func, "test_simple", module, externals=externals
+        )
+        loc = def_ir.computations[0].interval.loc
+        assert def_ir.computations[0].interval.start == gt_ir.AxisBound(
+            level=gt_ir.LevelMarker.START, offset=0, loc=loc
+        )
+        assert def_ir.computations[0].interval.end == gt_ir.AxisBound(
+            level=gt_ir.LevelMarker.START, offset=1, loc=loc
+        )
+
+    def test_none(self):
+        def definition_func(field: gtscript.Field[float]):
+            with computation(PARALLEL), interval(1, None):
+                field = 0
+
+        module = f"TestIntervalSyntax_none_{id_version}"
+        externals = {}
+        stencil_id, def_ir = compile_definition(
+            definition_func, "test_none", module, externals=externals
+        )
+        loc = def_ir.computations[0].interval.loc
+        assert def_ir.computations[0].interval.start == gt_ir.AxisBound(
+            level=gt_ir.LevelMarker.START, offset=1, loc=loc
+        )
+        assert def_ir.computations[0].interval.end == gt_ir.AxisBound(
+            level=gt_ir.LevelMarker.END, offset=0, loc=loc
+        )
+
+    def test_externals(self):
+        def definition_func(field: gtscript.Field[float]):
+            from __externals__ import kstart
+
+            with computation(PARALLEL), interval(kstart, -1):
+                field = 0
+
+        module = f"TestIntervalSyntax_externals_{id_version}"
+        for kstart in (3, gtscript.K[3]):
+            externals = {"kstart": kstart}
+            stencil_id, def_ir = compile_definition(
+                definition_func, "test_externals", module, externals=externals
+            )
+            loc = def_ir.computations[0].interval.loc
+            assert def_ir.computations[0].interval.start == gt_ir.AxisBound(
+                level=gt_ir.LevelMarker.START, offset=3, loc=loc
+            )
+            assert def_ir.computations[0].interval.end == gt_ir.AxisBound(
+                level=gt_ir.LevelMarker.END, offset=-1, loc=loc
+            )
+
+    def test_splitter(self):
+        def definition_func(field: gtscript.Field[float]):
+            from __gtscript__ import K
```

Comment:
Question: explicit importing K is still not needed, right?

Reply:
Actually, another question here: what does `K[2]` mean? It looks like `interval(K[2], -1)` means the same as `interval(2, -1)`, which I find confusing because it is a mix of the old "function-call" notation and the new slice notation.

I also think using the term splitter here is wrong since the resulting AxisBound is still an offset related to the implicit Axis.START level. Something similar in spirit to GridTools splitters would allow to write the AxisBound as `AxisBound(level=splitter, offset=0)` and then the syntax could be either `interval(splitter, -1)` or `interval(K[splitter:-1])` being splitter an int variable (external or local, or even run-time parameter if we allow this in the future).

Reply:
I can try to make a change that requires EITHER the old or new syntax to be used, but not either. Would that be a good approach?

Reply:
> I also think using the term splitter here is wrong since the resulting AxisBound is still an offset related to the implicit Axis.START level.

Good point. I'll rename the object that e.g. `gtscript.K[2]` returns to be a `gtscript._AxisOffset`.

Reply:
> what does K[2] mean? It looks like interval(K[2], -1) means the same as interval(2, -1), which I find confusing because it is a mix of the old "function-call" notation and the new slice notation.

I agree that mixing up the old and new style is confusing here. The technical detail is that if the `K[2]` were a an external instead of inlined (a `gtscript._AxisOffset` object), then the `ValueInliner` replaces this by an `ast.Constant` node before the `IRMaker` runs (which calls the `AxisIntervalParser`). That is to say that in the case of an external we cannot distinguish between the new and old style syntax.

I can add a check that in the case of old-style syntax AND not using named offsets (externals), then enforce that new and old style syntax is not mixed. I'll make that change now.